plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt

```diff
@@ -158,7 +158,7 @@ class AmazonQLanguageClientImpl(private val project: Project) : AmazonQLanguageC
             // The filepath sent by the server contains unicode characters which need to be
             // decoded for JB file handling APIs to be handle to handle file operations
             val fileToOpen = URLDecoder.decode(params.uri, StandardCharsets.UTF_8.name())
-            ApplicationManager.getApplication().invokeLater {
+            ApplicationManager.getApplication().invokeAndWait {
```

Comment:
```suggestion
        return CompletableFuture.supplyAsync(
            {
               ....
            },
            ApplicationManager.getApplication()::invokeLater
        )
```

Reply:
This still results in the race condition